### PR TITLE
Remove jQuery dependency

### LINKED ISSUE #175: Make a version with no dependencies
It depends on jQuery. Make a version with no dependencies at all. Would be sweet!

### PR DESCRIPTION
**Pull request by mrksbnch**
It should mostly be backwards-compatible, but there are some breaking changes:
- **No IE 8 support**: Although I haven't tested it yet, IE 9 should still work.
- **No ["programmatic value changes"](http://rangeslider.js.org/#js-example-change-value)**: There's just no way to detect `el.value` changes on an element. If you want to change the handle position, you'll have to call the `setPosition` method for now.

I haven't touched any of the actual calculations or styling parts for this pull request, but I think parts of them could be improved:
- Use CSS transformations instead of `left` and `width` to update the handle and fill elements
- Use `requestAnimationFrame` to throttle `resize` and `*move` events
- Cache some stuff like `getBoundingClientRect` etc.
- Not really sure about the difference between `getPositionFromNode` (internal) and `getBoundingClientRect`, but as far as I have seen it's more or less the same. Using only the latter, we could save some bytes and also tweak performance. **EDIT:** What is [this part](https://github.com/mrksbnch/rangeslider.js/blob/develop/src/rangeslider.js#L478) good for, anyway? I temporarily removed it and everything worked just fine. If we get rid of that part, we could also remove the `getPositionFromNode` function.
- Use `%` values instead of `px` values to position the handle. That way, you could, among other things, call the window `resize` event much less often.

Also, if this doesn't get merged, it's no big deal. I needed a small `input type="range"` "polyfill", anyway.

Resolves #175

### CODE REVIEW

**Comment by andreruffert** [comment]
@mrksbnch first of all thanks for the effort and the pr.
Give me some time to review and think about it ✌️

**Comment by avindra** [comment]
@mrksbnch : Awesome! You should probably rebase the commits to make the PR clean. There are lots of good docs floating around the web on how to do it.

TL;DR; Basically you just rebase, amend commit messages as necessary and force push, which will update the PR automatically.

**Comment by mrksbnch** [comment]
@avindra Good idea... done

**Comment by sasha-bichkov** [comment]
@mrksbnch, Hi!
Thank you for this PR! Unfortunately, I've came across a bug on iPad. I can't move "rangeslider__handle". To be more precise, it jumps to the beginning of the rangeslider and doesn't change its position anymore...

**Comment by mrksbnch** [comment]
@sasha-bichkov Thanks, I don't have an iPad here to test this, but it works on all of my other devices. Could you do me a favor and check if there's any error message in the console? If you are on a Mac, you can connect your iPad to your Mac and use Safari to check the console.

Also, please note that I haven't committed any files in the `dist` folder. So you'll have to run `npm run build` yourself before checking the examples on the `example/index.html` page.

**Comment by mrksbnch** [comment]
@sasha-bichkov I had some time and it occurred to me that I can just use the iOS Simulator on my Mac. So I did that and it turns out that you are right. Thanks again for reporting this!

Although I fixed and tested it, I'd highly appreciate it, if you could have another look on your iPad. It could be that the iOS Simulator is not 100% correct.

**Comment by mrmagooey** [comment]
I've been trying this pr out, and it seems like uglify is doing strange things to the constructor, i.e. adding in a `return` to the first statement in the constructor regardless of what that first statement is, which means that basically nothing in the constructor gets called. Is anyone else getting this?

**Comment by sasha-bichkov** [comment]
@mrksbnch I've just checked it out and it works perfect! I've tested on both an iPhone and an iPad. Thanks!

**Comment by mrksbnch** [comment]
@mrmagooey I just checked the example page with the minified JavaScript file and it works for me. So I'm not exactly sure why you are having this issue.

**Comment by atypicalprogrammer** [comment]
Has this been merged? Can we get a version of script without the jquery plugin code?

**Comment by lino-silva** [comment]
Hey everyone - great to check this thread has been going for a while! Is this being considered for merge any time soon?

**Comment by MohammedSiddiqui10p** [comment]
+1

**Comment by michalhudecek** [comment]
Is someone looking into the merge request? It would be awesome to have this done soon. Thanks!

**Comment by avindra** [comment]
@andreruffert can you please nominate a contributor as a new maintainer?

Seems like you dont have time for this project, which is cool, but there are folks still using it 😃

**Comment by mrksbnch** [comment]
I'm closing this pull request. I don't think it's worth resolving the conflicts considering that it'll probably never get merged.
